src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTable.cs

```diff
@@ -276,6 +276,28 @@ internal void AddEntity(IExternalEntity entity, DName displayName = default)
             }
 
             _variables.Add(entity.EntityName, nameInfo);
+        }
+
+        public void AddHostObject(string name, FormulaType type, Func<IServiceProvider, FormulaValue> getValue)

```

Comment:
We should document what getValue does on failure. Return null? Return an ErrorValue? Throw a CustomFunctionException (that we wrap and return an error)

Reply:
updated.